docs/design-docs/cli.md

```diff
@@ -0,0 +1,51 @@
+# Studio CLI
+
+## About this doc
+
+This document outlines the design and implementation details for the Studio CLI utility. It covers the high-level approach, data flows, and implementation details for this feature.
+
+## Context
+
+The Studio CLI (invoked with the `studio` command) is a globally available CLI utility allowing users to interact with various Studio features independently of the desktop application.
+
+## High level approach
+
+The CLI is independent of the main desktop app, but is written using mostly the same conventions. It's a node.js app written in Typescript, that's bundled into a single JS file using Webpack. Jest is used to test CLI modules in the same way as for regular Studio modules.
+
+To run the CLI, we first add a script to a directory on `$PATH`. This script runs the CLI JS file using the node runtime bundled with Studio. Running JS files independently of the main Studio app is possible thanks to the `ELECTRON_RUN_AS_NODE=1` option.
+
+The first iteration of the CLI shipped commands to create, read, update, and delete preview sites. To keep the business logic consolidated, we've refactored Studio to instantiate the CLI when creating, updating, and deleting preview sites.
+
+## Data flow
+
+1. When calling the CLI:
```

Comment:
Should we document the `appdata-v1.json` file, if I'm not wrong it's what corresponds to the "state of the CLI" tool. Where it lives...

Reply:
This is the main Studio config file. I've tweaked the wording to clarify this.

Reply:
Not disagreeing that the more we develop the CLI, the more the needs around the config file(s) will change. However, there are no concrete plans here yet, so I agree about waiting with documenting that part.